File: corgi-dm/src/main/scala/com/dyingbleed/corgi/dm/LivyJob.scala
package com.dyingbleed.corgi.dm

import org.apache.livy.{Job, JobContext}

/**
  * Created by 李震 on 2019/3/12.
  */
class LivyJob(args: Array[String]) extends Job[Unit] {

  override def call(ctx: JobContext): Unit = {
    Application.execute(ctx.sparkSession(), args)
  }

}

File: corgi-dm/src/main/java/com/dyingbleed/corgi/dm/LivyJob.java
package com.dyingbleed.corgi.dm;

import org.apache.livy.Job;
import org.apache.livy.JobContext;

/**
 * Created by 李震 on 2019/3/21.
 */
public class LivyJob implements Job<Void> {

    private String[] args;

    public LivyJob(String[] args) {
        this.args = args;
    }

    @Override
    public Void call(JobContext ctx) throws Exception {
        Application.execute(ctx.sparkSession(), this.args);
        return null;
    }

}
